src/UI/Windows/BigListBox.cs

```diff
@@ -50,39 +49,30 @@ public class BigListBox<T>
     public BigListBox(
         string prompt,
         string okText,
-        bool addSearch,
+        in bool addSearch,
```

Comment:
Is there a reason why we need the `in` keyword for this method?

Reply:
Need? No. Small efficiency gain, and callers don't have to specify it, so it's a non-breaking change.

Reply:
Expanding on that...

I generally try to put `in` on all value type parameters. There's no downside and it really should have been default behavior. At the native level, it avoids one memory copy in all situations, and, in cases where the parameter is never used by the callee for assignment to anything else (like just being used for an if statement or something), the value is never copied at all.

It also allows the compiler to make additional optimizations, in some cases, since it's a mathematical restriction on the program, as a whole.

One neat thing you gain from it, for free, is that you're prohibited from attempting to mutate the parameter, when it's a value type, because in means the reference cannot be reassigned, and changing a value type means changing that. So it prevents a subtle source of potential bugs since, otherwise, you can operate on the parameter, but, being a value type, you've copied it and are now no longer touching the original.

The fact that callers don't have to specify `in` makes it even nicer, since it still gets called that way anyway. It's optional at the call site.

I honestly wish c# treated value type parameters that way by default.

Plus, if you ever switch to a more complex value type, in its place, you already have it there and reap significant benefits from not passing a large struct around by value.

In another project I created, I went back and changed most value type parameters to `in` (and some immutable ref types, too, like strings) and the results were tangible to a degree I did not expect, both in memory working set of the application and in CPU usage. The unit tests even sped up noticeably.